### ISSUE TITLE
Monster remove ladder

### ISSUE THREAD

**Issue by peonso** [TwistedScorpio/OTHire#72]
I do not know if it's a bug but from what I tested here, a monster removed a ladder on their way
`itemid 1386 a ladder` 

[before](https://s16.postimg.org/qkd7b2ogl/before0001.png)

[after](https://s16.postimg.org/hbb10yfkl/after001.png)

@diegorodriguesvieira

**Comment by diegorodriguesvieira** [comment]
![image](https://media.giphy.com/media/l0HlNp0Tuu4ecj7mE/giphy.gif)

**Comment by diegorodriguesvieira** [comment]
Can anyone help me find this bug in OTClient? :/

**Comment by peonso** [comment]
https://github.com/edubart/otclient/issues/597
https://github.com/edubart/otclient/issues/418
https://github.com/edubart/otclient/issues/410
https://otland.net/threads/7-x-stairs-ladders-disappear-if-when-i-kill-2-monsters-there.239617/

**Comment by diegorodriguesvieira** [comment]
Thanks man, I'll try to fix it...

**Comment by diegorodriguesvieira** [comment]
I fix it, I don't know if it's the best solution...
Open tile.cpp and find: 
`g_game.FreeThing(oldSplash);`

https://github.com/TwistedScorpio/OTHire/blob/master/source/tile.cpp#L921

And add below:
`onUpdateTile();`

**Comment by Anastaciaa** [comment]
This is not properly fix. It causes some screen jumps while getting hits.

**Comment by peonso** [comment]
@Anastaciaa in both OTClient and default one?

**Comment by Anastaciaa** [comment]
Yes

**Comment by diegorodriguesvieira** [comment]
@Anastaciaa How can I fix it?

**Comment by Anastaciaa** [comment]
@diegorodriguesvieira send me your skype ill contact you. I have to look on this through team viewer because it's abit complicated.

**Comment by diegorodriguesvieira** [comment]
Thank you @Anastaciaa  for the help! My skype: diego.ldn

**Comment by Riverlance** [comment]
https://github.com/edubart/otclient/issues/790

**Comment by don-romano** [comment]
**_TFS 1.3 SOLUTION_ FOR TIBIA 7.4 - 7.72**

Open vc14/theforgottenserver.vcxproj

Open tile.cpp, hit CTRL + F and look for "//remove old splash if exists"

Above this phrase add
`SpectatorHashSet spectators;`
`g_game.map.getSpectators(spectators, getPosition(), true, true);`

Below _postRemoveNotification(oldSplash, nullptr, 0);_ add
`onUpdateTile(spectators);`

**Comment by Felipe21** [comment]
@don-romano  i tested your code , but the problem is that after the monter dies  if i want to use the ladder, it doesn't allow/let me use it... there is a missing code , any ideas?
`
```
				SpectatorVec spectators; //7.4 sourceedit fix splash //7.4 sourceedit
				g_game.map.getSpectators(spectators, getPosition(), true, true);//7.4 sourceedit fix splash //7.4 sourceedit
				//remove old splash if exists
				for (ItemVector::const_iterator it = items->getBeginTopItem(), end = items->getEndTopItem(); it != end; ++it) {
					Item* oldSplash = *it;
					if (!Item::items[oldSplash->getID()].isSplash()) {
						continue;
					}

					removeThing(oldSplash, 1);
					oldSplash->setParent(nullptr);
					g_game.ReleaseItem(oldSplash);
					postRemoveNotification(oldSplash, nullptr, 0);
					onUpdateTile(spectators); //7.4 sourceedit fix splash //7.4 sourceedit
					break;`
```

@peonso
